packages/framework/src/Model/Order/Withdrawal/WithdrawalChecker.php

```diff
@@ -18,11 +18,13 @@ class WithdrawalChecker
      * @param \Shopsys\FrameworkBundle\Model\Order\Withdrawal\WithdrawalDeadlineCalculation $withdrawalDeadlineCalculation
      * @param \Shopsys\FrameworkBundle\Model\Order\OrderFacade $orderFacade
      * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \Shopsys\FrameworkBundle\Model\Order\Withdrawal\WithdrawalRequestRepository $withdrawalRequestRepository
      */
     public function __construct(
         protected readonly WithdrawalDeadlineCalculation $withdrawalDeadlineCalculation,
         protected readonly OrderFacade $orderFacade,
         protected readonly Domain $domain,
+        protected readonly WithdrawalRequestRepository $withdrawalRequestRepository,
```

Comment:
I would expect facade here tbh. Is a reason to use directly a repository?

Reply:
yes, originally, there was a facade, but this was another place where the circular reference occurs (`WithdrawalRequestFacade` has a dependency on `WihdrawalChecker`, it uses it in `createWithdrawalRequest` method, and the checker needs the facade to find the withdrawal by order :melting_face: Another (and maybe better) solution would be splitting `WithdrawalRequestFacade` into two services - one for requests creating, another for fetching them :shrug: But at this point, I am totaly ok with using the repository within the checker :smile: